### PR TITLE
Creating index with empty "name"/"ddoc" should return 400

### PR DESCRIPTION
**Pull request by jiahuili430**
## Overview
Creating an index with `"ddoc":""` or `"name":""` should return a `400 Bad Request`.

## Testing recommendations
```bash
curl -X DELETE $db/abc && curl -X PUT $db/abc
curl -X POST $db/abc/_index -H 'Content-Type: application/json' -d '{"ddoc": "", "index": {"fields": ["foo"]}}'
curl -X POST $db/abc/_index -H 'Content-Type: application/json' -d '{"name": "", "index": {"fields": ["bar"]}}'

{"error":"invalid_empty_string","reason":"Index name/ddoc cannot be empty"}
```

## Related Issues or Pull Requests
https://github.com/apache/couchdb/issues/1472

## Checklist

- [x] Code is written and works correctly
- [x] Changes are covered by tests
- [ ] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [ ] A PR for documentation changes has been made in https://github.com/apache/couchdb-documentation

### CODE REVIEW

File: src/mango/src/mango_opts.erl [hunk 1]
```diff
@@ -240,6 +241,13 @@ validate_idx_name(auto_name) ->
 validate_idx_name(Else) ->
     is_string(Else).
 
+validate_idx_ddoc(auto_name) ->
```
[the remark this answers is not in the document]
Comment (jiahuili430) [reply src/mango/src/mango_opts.erl]: Changed, thanks for the review.

File: src/mango/test/01-index-crud-test.py [hunk 2]
```diff
@@ -68,17 +68,17 @@ def test_bad_types(self):
             else:
                 raise AssertionError("bad create index")
 
-    def test_bad_names(self):
-        bad_names = [True, False, 1.5, {"foo": "bar"}, [None, False]]
-        for bn in bad_names:
+    def test_bad_names_ddocs(self):
+        bad_args = ["", True, False, 1.5, {"foo": "bar"}, [None, False]]
```
Comment (tonysun83) [inline src/mango/test/01-index-crud-test.py:72]: this only tests for `"name": ""` ? Can we add a testcase where `"ddoc": ""` can it receives the same error? The original bug has `"ddoc" : ""`

File: src/mango/src/mango_error.erl [hunk 3]
```diff
@@ -266,11 +266,11 @@ info(mango_opts, {invalid_selector_json, BadSel}) ->
...
         fmt("Selector must be a JSON object, not: ~w", [BadSel])
     };
-info(mango_opts, {invalid_index_name, BadName}) ->
+info(mango_opts, invalid_empty_string) ->
     {
         400,
-        <<"invalid_index_name">>,
-        fmt("Invalid index name: ~w", [BadName])
+        <<"invalid_empty_string">>,
+        <<"Index name/ddoc cannot be empty">>
```
Comment (tonysun83) [inline src/mango/src/mango_error.erl:273]: <<"Index name or ddoc cannot be empty string">>

File: src/mango/test/01-index-crud-test.py (hunk 2 above)
Comment (jiahuili430) [reply src/mango/test/01-index-crud-test.py]: Added another test case [test_bad_ddocs()](https://github.com/apache/couchdb/pull/3990/files#diff-243cff897c08c3a4b060fed2dc84a22d35740500fbf075e6209ba5013f0fd7faR81), thanks for the review.

**Review by tonysun83 — APPROVED** [review]

**Pull request merged**